crates/chain/src/chain_data.rs

```diff
@@ -247,23 +247,25 @@ impl<A: Anchor> FullTxOut<ObservedAs<A>> {
     /// This is the alternative version of [`is_mature`] which depends on `chain_position` being a
     /// [`ObservedAs<A>`] where `A` implements [`Anchor`].
     ///
+    /// Depending on the implementation of [`confirmation_height_upper_bound`] in [`Anchor`], this
+    /// method may return false-negatives. In other words, interpretted confirmation count may be
+    /// less than the actual value.
+    ///
     /// [`is_mature`]: Self::is_mature
+    /// [`confirmation_height_upper_bound`]: Anchor::confirmation_height_upper_bound
     pub fn is_mature(&self, tip: u32) -> bool {
```

Comment:
My personal opinion is it's not prudent to ship with this. This breaks coinbase and time-locked transaction's maturity calc, and to me, it feels big enough to be addressed. 

Another problem is if the current definition of `Anchor` allows them to be updated to new heights as the blockchain progresses, this will **recurringly** break the maturity determination of locked transactions.   

If we expect this to be fixed by users by using custom `Anchor`, then let us just fix it here because we **are** the first "user" of this lib. 

There is also not one single way to solve this, it can be done in many ways.

My proposal:
 - Define `Anchor` as the inclusion `BlockId` of the transaction. It has two methods `inclusion_block()`, and `inclusion_height()`. (feel free to choose other names).

 - `impl Anchor for BlockId` is basically the two getters into `BlockId`. That's the minimal definition of an `Anchor`. 

 - Only change `Anchor`s in `TxGraph` in case of reorgs. 

 - Explain in the `struct Anchor` doc how they can be implemented on any custom structure to store other metadata of txs, ex: confirmation time, Merkle tree position of tx, etc.
 
 That, in my mind, provides the MVP for `Anchor` without any major issues and satisfies its purpose. Others can have other ideas too.

Reply:
ACK on team discussion to add this via another trait to include confirmation height.